libacb/finish.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2015 The appc Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package libacb
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/appc/acbuild/libacb/util"
+
+	"github.com/appc/spec/schema/types"
+)
+
+func Finish(actool, tmpaci, output, depstore, scratchpath string, overwrite bool) error {
+	man, err := util.GetManifest(tmpaci)
+	if err != nil {
+		return err
+	}
+
+	if man.App != nil && testEq(man.App.Exec, placeholderexec) {
+		return fmt.Errorf("can't finish build, run command was never set")
+	}
+
+	if man.Name == types.ACIdentifier("") {
+		return fmt.Errorf("can't finish build, name was never set")
+	}
+
+	ex, err := util.Exists(output)
+	if err != nil {
+		return err
+	}
+	if ex {
+		if !overwrite {
+			return fmt.Errorf("ACI already exists: %s", output)
+		}
+		err := os.Remove(output)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = util.Exec(actool, "build", "--overwrite", tmpaci, output)
+	if err != nil {
```

Comment:
what about using appc/spec/aci ImageWriter instead of calling an external process?